cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/statements/ForEachStatement.kt

```diff
@@ -47,7 +46,7 @@ class ForEachStatement : LoopStatement(), BranchingNode, StatementHolder {
         astOptionalEdgeOf<Statement>(
             onChanged = { _, new ->
                 val end = new?.end
-                if (end is Reference) {
+                if (end is HasAccess) {
```

Comment:
Could be written as `(end as? HasAccess)?.access = ....` to be consistent with the other assignments.